attestation/k8smanifest/k8s.go

```diff
@@ -0,0 +1,720 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8smanifest
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/in-toto/go-witness/attestation"
+	"github.com/in-toto/go-witness/cryptoutil"
+	"github.com/in-toto/go-witness/internal/jsoncanonicalizer"
+	"github.com/in-toto/go-witness/log"
+	"github.com/in-toto/go-witness/registry"
+	"github.com/invopop/jsonschema"
+	"gopkg.in/yaml.v3"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Name is the identifier for this attestor.
+const Name = "k8smanifest"
+
+// Type is the URI identifying the predicate type.
+const Type = "https://witness.dev/attestations/k8smanifest/v0.2"
+
+// RunType is the run stage at which this attestor is executed.
+const RunType = attestation.PostProductRunType
+
+// Default ephemeral fields to remove.
+var defaultEphemeralFields = []string{
+	"metadata.resourceVersion",
+	"metadata.uid",
+	"metadata.creationTimestamp",
+	"metadata.managedFields",
+	"metadata.generation",
+	"status",
+}
+
+// Default ephemeral annotations to remove.
+var defaultEphemeralAnnotations = []string{
+	"kubectl.kubernetes.io/last-applied-configuration",
+	"deployment.kubernetes.io/revision",
+	"witness.dev/content-hash",
+	"cosign.sigstore.dev/message",
+	"cosign.sigstore.dev/signature",
+	"cosign.sigstore.dev/bundle",
+}
+
+// RecordedObject stores ephemeral-cleaned doc details.
+type RecordedObject struct {
+	FilePath       string               `json:"filePath"`
+	Kind           string               `json:"kind"`
+	Name           string               `json:"name"`
+	Data           json.RawMessage      `json:"data"`
+	CanonicalJSON  string               `json:"canonicalJSON"`
+	SubjectKey     string               `json:"subjectKey"`
+	ComputedDigest cryptoutil.DigestSet `json:"computedDigest"`
+	RecordedImages []RecordedImage      `json:"recordedImages"`
+}
+
+type ClusterInfo struct {
+	Server string `json:"server"`
+}
+
+// Recorded image stores the details of images found in kubernetes manifests
+type RecordedImage struct {
+	Reference string            `json:"reference"`
+	Digest    map[string]string `json:"digest"`
+}
+
+// Attestor implements the Witness Attestor interface for Kubernetes manifests.
+type Attestor struct {
```

Comment:
We should change all of the jsontags to be consistent with the rest of our attestors. Just in this one struct we have snake case, upper camel case, and lower camel case. In other attestors we tend to prefer `ignorefields` and `ignoreannotations`, but I think `ignoreFields` and `ignoreAnnotations` is also acceptable.

Reply:
Sorry, didn't notice this. I will fix.